zerver/tests/tests.py

```diff
@@ -586,6 +601,47 @@ def test_add_bot(self):
         bot = bots[0]
         self.assertEqual(bot['bot_owner'], 'hamlet@zulip.com')
 
+    def test_add_bot_with_username_in_use(self):
+        # type: () -> None
+        self.login("hamlet@zulip.com")
+        self.assert_num_bots_equal(0)
+        result = self.create_bot()
+        self.assert_num_bots_equal(1)
+
+        bot_info = dict(
+            full_name='whatever',
+            short_name='hambot',
+            )
+        result = self.client.post("/json/bots", bot_info)
+        self.assert_json_error(result, 'Username already in use')
+
+    def test_add_bot_with_user_avatar(self):
+        # type: () -> None
+        self.login("hamlet@zulip.com")
+        self.assert_num_bots_equal(0)
+        with open(os.path.join(TEST_AVATAR_DIR, 'img.png'), 'rb') as fp:
+            result = self.create_bot(file=fp)
+        self.assert_num_bots_equal(1)
+
+        profile = get_user_profile_by_email('hambot-bot@zulip.com')
+        self.assertEqual(profile.avatar_source, UserProfile.AVATAR_FROM_USER)
+
+    def test_add_bot_with_too_many_files(self):
+        # type: () -> None
+        self.login("hamlet@zulip.com")
+        self.assert_num_bots_equal(0)
+        with open(os.path.join(TEST_AVATAR_DIR, 'img.png'), 'rb') as fp1:
+            with open(os.path.join(TEST_AVATAR_DIR, 'img.gif'), 'rb') as fp2:
```

Comment:
you can write this as `with open(..) as fp1, open(...) as fp2` if you don't want the extra indent.  Not sure it's better, just wanted to mention that as an option.